tools/topology/topology2/cavs-rt5682-ssp0spk.conf

```diff
@@ -0,0 +1,291 @@
+<searchdir:include>
+<searchdir:include/common>
+<searchdir:include/components>
+<searchdir:include/dais>
+<searchdir:include/pipelines/cavs>
+<searchdir:platform/intel>
+
+<vendor-token.conf>
+<manifest.conf>
+<pdm_config.conf>
+<tokens.conf>
+<virtual.conf>
+<passthrough-playback.conf>
+<passthrough-capture.conf>
+<passthrough-be.conf>
+<host-copier-gain-mixin-playback.conf>
+<mixout-gain-dai-copier-playback.conf>
+<deepbuffer-playback.conf>
+<data.conf>
+<pcm.conf>
+<pcm_caps.conf>
+<fe_dai.conf>
+<ssp.conf>
+<dmic.conf>
+<hda.conf>
+<hw_config.conf>
+<manifest.conf>
+<route.conf>
+<common_definitions.conf>
+<copier.conf>
+<pipeline.conf>
+<dai.conf>
+<host.conf>
+<dmic-default.conf>
+<hdmi-default.conf>
+
+Define {
+	MCLK 				24576000
+	NUM_DMICS			0
+	# override DMIC default definitions
+	NUM_DMICS			0
+	DMIC0_ID			1
+	DMIC1_ID			2
+	DMIC0_HOST_PIPELINE_ID		13
+	DMIC0_DAI_PIPELINE_ID		14
+	DMIC0_HOST_PIPELINE_SINK	'copier.host.13.1'
+	DMIC0_DAI_PIPELINE_SRC		'copier.DMIC.14.1'
+	DMIC0_NAME			'dmic01'
+	DMIC0_PCM_CAPS			'Passthrough Capture 13'
+	DMIC0_PIPELINE_STREAM_NAME	'copier.DMIC.14.1'
+	# override BE link ids
+	NUM_HDMIS			0
+	HDMI1_ID			3
+	HDMI2_ID			4
+	HDMI3_ID			5
+	HDMI4_ID			6
+	SPK_ID				7
+	PLATFORM 			"none"
+	USE_CHAIN_DMA			"false"
+	DEEP_BUFFER_PIPELINE_ID		15
+	DEEP_BUFFER_PCM_ID		31
+	DEEP_BUFFER_PIPELINE_SRC	'mixin.15.1'
+	DEEP_BUFFER_PIPELINE_SINK	'mixout.2.1'
+	DEEP_BUFFER_PCM_NAME		'Deepbuffer Jack Out'
+}
+
+# override defaults with platform-specific config
+IncludeByKey.PLATFORM {
+	"mtl"	"platform/intel/mtl.conf"
+}
+
+# include DMIC config if needed.
+IncludeByKey.NUM_DMICS {
+	"[1-4]"	"platform/intel/dmic-generic.conf"
+}
+
+# include HDMI config if needed.
+IncludeByKey.NUM_HDMIS {
+	"[3-4]"	"platform/intel/hdmi-generic.conf"
+}
+
+# include deep buffer config if buffer size is in 1 - 1000 ms.
+IncludeByKey.DEEPBUFFER_FW_DMA_MS{
+	"[1-1000]" "platform/intel/deep-buffer.conf"
+}
+#
+# List of all DAIs
+#
+#SSP Index: 0, Direction: duplex
+Object.Dai {
+	SSP."0" {
+		id 		$SPK_ID
+		dai_index	0
+		direction	"playback"
+		name		SSP0-Codec
+		default_hw_conf_id	0
+		sample_bits		32
+		io_clk		$MCLK
+
+		Object.Base.hw_config."SSP0" {
+			id	0
+			mclk_freq	$MCLK
+			bclk_freq	3072000
+			tdm_slot_width	32
+		}
+	}
+	SSP."2" {
+		id 		0
+		dai_index	2
+		direction	"duplex"
+		name		SSP2-Codec
+		default_hw_conf_id	0
+		sample_bits		32
+		io_clk		$MCLK
+
+		Object.Base.hw_config."SSP2" {
+			id	0
+			mclk_freq	$MCLK
+			bclk_freq	3072000
+			tdm_slot_width	32
+		}
+	}
+}
+
+#
+# Pipeline definitions
+#
+# PCM0 ---> gain ----> Mixin ----> Mixout ----> gain ----> SSP0
+# PCM1 ---> gain ----> Mixin ----> Mixout ----> gain ----> SSP2
+#
+# SSP0 ----> PCM0
+# SSP2 ----> PCM1
```

Comment:
this is not compatible with MTL requirements of having multiple capture streams, so this looks like an early version and certainly not what will be productized.